pyvista/core/filters/poly_data.py

```diff
@@ -3892,3 +3897,111 @@ def protein_ribbon(self, progress_bar=False):
         alg.SetInputData(self)
         _update_alg(alg, progress_bar, "Generating Protein Ribbons")
         return _get_output(alg)
+
+    def generate_labelmap(
```

Comment:
Looking for feedback on the name. The term `labelmap` is kind of domain specific to medical images, maybe something more general is better?

This method is basically the inverse surface nets from `contour_labeled`.  So we could call it something like `contour_to_labeled` to better relate the two? 

Or, with #5968, we also have the option to name _both_ filters, e.g. rename `contour_labeled` -> `image_labels_to_polydata`, and name the filter from this PR `polydata_to_image_labels`

Or, even still, we could better align the name with `voxelize` and `voxelize_volume`, and maybe call this filter `voxelize_labels`?

Reply:
Coming from material sciences, `labelmap` is not the usual terminology, although some literature uses `labelling`.

`voxelize` is more common in this field but does not carry the idea that voxels are labeled. Another one could be `segmentation`.

Reply:
Yes I think the term `label` is perhaps a bit vague/non-obvious outside of specific fields. Maybe `voxelize_binary_mask`  is good? The method currently only works for binary masks anyway, I'm thinking a multi-label version is probably better suited for a separate filter (and separate PR), maybe call that one `voxelize_label_mask`